numba/cuda/tests/cudapy/test_atomics.py

```diff
@@ -193,6 +193,78 @@ def atomic_sub_double_global_3(ary):
                               cuda.atomic.sub, atomic_cast_to_uint64)
 
 
+def atomic_and(ary, op2):
+    atomic_binary_1dim_shared(ary, ary, op2, uint32, 32,
+                              cuda.atomic.and_, atomic_cast_none, 1)
+
+
+def atomic_and2(ary, op2):
+    atomic_binary_2dim_shared(ary, op2, uint32, (4, 8),
+                              cuda.atomic.and_, atomic_cast_none)
+
+
+def atomic_and3(ary, op2):
+    atomic_binary_2dim_shared(ary, op2, uint32, (4, 8),
+                              cuda.atomic.and_, atomic_cast_to_uint64)
+
+
+def atomic_and_global(idx, ary, op2):
+    atomic_binary_1dim_global(ary, idx, 32, op2, cuda.atomic.and_)
+
+
+def atomic_and_global_2(ary, op2):
+    atomic_binary_2dim_global(ary, op2, cuda.atomic.and_,
+                              atomic_cast_none)
+
+
+def atomic_or(ary, op2):
+    atomic_binary_1dim_shared(ary, ary, op2, uint32, 32,
+                              cuda.atomic.or_, atomic_cast_none, 0)
+
+
+def atomic_or2(ary, op2):
+    atomic_binary_2dim_shared(ary, op2, uint32, (4, 8),
+                              cuda.atomic.or_, atomic_cast_none)
+
+
+def atomic_or3(ary, op2):
+    atomic_binary_2dim_shared(ary, op2, uint32, (4, 8),
+                              cuda.atomic.or_, atomic_cast_to_uint64)
+
+
+def atomic_or_global(idx, ary, op2):
+    atomic_binary_1dim_global(ary, idx, 32, op2, cuda.atomic.or_)
+
+
+def atomic_or_global_2(ary, op2):
+    atomic_binary_2dim_global(ary, op2, cuda.atomic.or_,
+                              atomic_cast_none)
+
+
+def atomic_xor(ary, op2):
+    atomic_binary_1dim_shared(ary, ary, op2, uint32, 32,
+                              cuda.atomic.xor, atomic_cast_none, 0)
+
+
+def atomic_xor2(ary, op2):
+    atomic_binary_2dim_shared(ary, op2, uint32, (4, 8),
+                              cuda.atomic.xor, atomic_cast_none)
+
+
+def atomic_xor3(ary, op2):
+    atomic_binary_2dim_shared(ary, op2, uint32, (4, 8),
+                              cuda.atomic.xor, atomic_cast_to_uint64)
+
+
+def atomic_xor_global(idx, ary, op2):
+    atomic_binary_1dim_global(ary, idx, 32, op2, cuda.atomic.xor)
+
+
+def atomic_xor_global_2(ary, op2):
+    atomic_binary_2dim_global(ary, op2, cuda.atomic.xor,
+                              atomic_cast_none)
+
+
```

Comment:
Can probably just generate this closing over `op2`? but equally fine to leave as is.